core/domain/base_model_validators.py

```diff
@@ -379,6 +382,25 @@ def validate(cls, item):
         for func in cls._get_external_instance_custom_validation_functions():
             func(item, cls.field_name_to_external_model_references)
 
+    @classmethod
+    def validate_deleted(cls, item):
```

Comment:
Do we need a similar validation check for UserQueryModels to be marked deleted after 4 weeks?

Reply:
Do we also have UserQueryModelValidator._validate_models_marked_deleted?

Reply:
No, but yes. The thing for any model that is marked as deleted only the `validate_deleted` is run, which checks that they are hard-deleted after 8 weeks. If the model is not marked as deleted (and it is `UserQueryModel`) the `validate is run, meaning that for `UserQueryModel` the `_validate_old_models_are_deleted` is run which checks that there are no models older than 4 weeks. So the `_validate_old_models_are_deleted` tehcnically checks if the model is marked as deleted and then later `validate_deleted` checks that the model is hard deleted. But I think it would make sense to rename `_validate_old_models_are_deleted` to `_validate_old_models_marked_deleted`.